lib/conjur/conjur_config.rb

```diff
@@ -26,6 +26,30 @@ class ConjurConfig < Anyway::Config
       authenticators: []
     )
 
+    def initialize(*args)
+      super(*args)
+
+    # If theh config file is not a valid YAML document, we want
+    # to raise a user-friendly ConfigValidationError rather than
+    # a Psych library exception.
+    rescue Psych::SyntaxError => e
+      raise(
+        ConfigValidationError,
+        "Config file contains a YAML syntax error: #{e.message}"
+      )
+
+    # If the config file is valid YAML, but the root object is
+    # not a YAML dictionary, this raises one of a number of
+    # NoMethodError exceptions because AnywayConfig assumes parsing
+    # the config file will result in a Ruby Hash. We capture
+    # this and raise a more user-friendly error message.
+    rescue NoMethodError
+      raise(
+        ConfigValidationError,
+        "Config file is not a valid YAML dictionary."
```

Comment:
It's (probably) possible that a `NoMethodError` can arise for a different reason since it's so generic. Would it make more sense to have an error message that's less certain about the cause? Something like "Unable to parse config file. Please ensure that it is a valid YAML dictionary."

Reply:
Yeah, that makes sense. I'll make the change.